app/Controllers/PostControllers.php

```diff
@@ -0,0 +1,300 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controllers;
+
+
+use App\View;
+use App\Models\Post;
+use App\Models\User;
+use App\Models\Comment;
+
+class PostControllers
+{
+    protected $postModel;
+
+    public function __construct()
+    {
+        $this->postModel = new Post();
+        $this->userModel = new User();
+        $this->commentModel = new Comment();
+    }
+
+    //Get All Blog Posts
+    public function blogPosts()
+    {
+        $posts = $this->postModel->getAllBlogPosts();
+        $data = [
+            'posts' => $posts,
+        ];
+        return View::make('/blogs/home', $data);
+    }
+
+    //admin homepage see blogs
+    public function adminSeeBlogs()
+    {
+        $posts = $this->postModel->adminBlogInfoHome();
+        $blogData = [
+            'posts' => $posts
+        ];
+        return View::make('/admin/home', $blogData);
+    }
+
+    public function addBlog()
+    {
+        if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+            $_POST = filter_input_array(INPUT_POST, FILTER_SANITIZE_STRING);
+            $data = [
+                'title' => trim($_POST['title']),
+                'blog_body' => trim($_POST['blog_body']),
+                'user_id' => $_SESSION['user_id'],
+                'title_err' => '',
+                'blog_body_err' => '',
+            ];
+            //VALIDATE data
+            if (empty($data['title'])) {
+                $data['title_err'] = 'Please enter a blog title';
+            }
+            if (empty($data['blog_body'])) {
```

Comment:
Maybe create a Helper function to process all the input validation errors, assigning the error message to the corresponding variables